### PR TITLE
<cmd>+b should toggle breakpoints

### PR DESCRIPTION
**Pull request by arthur801031**
#1112

### CODE REVIEW

**Comment by jasonLaster** [comment]
Hey @arthur801031 this looks great. I'm glad that codeMirror.getCursor() did the trick

I think, the `shortcuts.on` handlers passing shortcut in is kind of a smell. so I investigated adding some anonymous functions and they seemed to work really well. My big concern was that `shortcuts.off` would not release the handler, but it also seemed to work and remove all the listeners.

Here's a patch where I cleaned up your handlers and the editor search bar. If you'd like, it'd be great to have a new PR that cleans up those handlers too!

https://gist.github.com/jasonLaster/2b9d515ea0c134c3d1cfa5f72189c871

Thanks for all the hard work this week. You're killing it!

**Comment by arthur801031** [comment]
Hi @jasonLaster

Is the new PR for cleaning up handlers in different files? Or are you referring to the handlers in EditorSearchBar.js? Thank you very much for your help and feedback. Have a great Thanksgiving! :)

File: src/components/Editor.js [hunk 1]
```diff
@@ -127,6 +138,10 @@ const Editor = React.createClass({
   },
 
   toggleBreakpoint(line) {
+    if (!line) {
```
Comment (jasonLaster) [inline src/components/Editor.js]: Hmm do you need to check for a line?

File: src/components/Editor.js [hunk 2]
```diff
@@ -251,7 +266,9 @@ const Editor = React.createClass({
       label: cbLabel,
       accesskey: "C",
       disabled: false,
-      click: () => this.showConditionalPanel(line)
+      click: () => this.showConditionalPanel(
+        getCursorLine(this.editor.codeMirror)
```
Comment (jasonLaster) [inline src/components/Editor.js]: Don't you have line already?

**Comment by arthur801031** [comment]
Hi @jasonLaster  

I've made changes based on your comments. Please check. Thank you! :)

**Comment by jasonLaster** [comment]
Thanks @arthur801031!

**Pull request merged**